src/librustc_macros/src/tt.rs

```diff
@@ -0,0 +1,65 @@
+use proc_macro2::{Delimiter, TokenStream, TokenTree};
+
+pub struct TT<'a>(pub &'a TokenTree);
```

Comment:
This code was extracted from `syn`. Let's make sure the licensing works out.

Reply:
Can't we just use `syn` as a dependency here?

Reply:
The code is private in `syn`.

Reply:
That seems resolvable :laughing: 	Please attempt to make the code public in syn (or public behind a cargo feature that also turns on something unstable as to not stabilize this if making it public is undesirable right now).

Reply:
@dtolnay Do you know of a way to parse just doc comments? So I could get rid of this code?

Reply:
Yes, I suggested an easier way at the place this is used.